core/src/main/java/org/elasticsearch/index/percolator/PercolatorFieldMapper.java

```diff
@@ -152,8 +174,37 @@ public PercolatorFieldMapper(String simpleName, MappedFieldType fieldType, Mappe
     }
 
     @Override
+    public FieldMapper updateFieldType(Map<String, MappedFieldType> fullNameToFieldType) {
+        PercolatorFieldMapper updated = (PercolatorFieldMapper) super.updateFieldType(fullNameToFieldType);
+        KeywordFieldMapper queryTermsUpdated = (KeywordFieldMapper) queryTermsField.updateFieldType(fullNameToFieldType);
+        KeywordFieldMapper unknownQueryUpdated = (KeywordFieldMapper) unknownQueryField.updateFieldType(fullNameToFieldType);
+        BinaryFieldMapper queryBuilderUpdated = (BinaryFieldMapper) queryBuilderField.updateFieldType(fullNameToFieldType);
+        if (updated == this && queryTermsUpdated == queryTermsField && unknownQueryUpdated == unknownQueryField &&
+                queryBuilderUpdated == queryBuilderField) {
+            return this;
+        }
+        if (updated == this) {
+            updated = (PercolatorFieldMapper) updated.clone();
+        }
+        updated.queryTermsField = queryTermsUpdated;
+        updated.unknownQueryField = unknownQueryUpdated;
+        updated.queryBuilderField = queryBuilderUpdated;
+        return updated;
+    }
```

Comment:
scratch last comment, I thought this was referring to a different piece of code... 

A test was failing, when adding a second percolator field with the same name, mapping identity assertions were failing:
MapperService#assertMappersShareSameFieldType()

But if we restrict that there can only be one percolator field type per index then we don't need to method any more.